### PR TITLE
Implement NoArgument cop to detect default arguments

### LINKED ISSUE #4: Prohibit the use of magic numbers in keyword arguments
## What 

Do not allow magic numbers to be used as default values in keyword arguments

``` ruby 
# Offence
def example_method(number_of_iterations: 2)
end

# No offence
def example_method(number_of_iterations: DEFAULT_ITERATIONS_PER_THING)
end
```

## Why

Prefer descriptive constants to describe default values

### LINKED ISSUE #38: - `NoArguments`

### PR DESCRIPTION
**Pull request by AgentAntelope**
fixes #38
fixes #4

### CODE REVIEW

File: lib/rubocop/cop/magic_numbers/no_argument.rb [hunk 1]
```diff
@@ -1,11 +1,88 @@
...
+      #
+      # bad: def on_the_wall(bottles = 100)
+      #
+      # good: def on_the_wall(bottles = DEFAULT_BOTTLE_COUNT)
+      class NoArgument < Base
+        MAGIC_NUMBER_OPTIONAL_ARGUMENT_PATTERN = <<-PATTERN
+          (def
+            _
+            (args
+              <({kwoptarg optarg}
```
Comment (Bodacious) [inline lib/rubocop/cop/magic_numbers/no_argument.rb:19]: lol

File: test/rubocop/cop/magic_numbers/no_argument/method_test.rb
Comment (Bodacious) [inline test/rubocop/cop/magic_numbers/no_argument/method_test.rb:1]: Do we still need this file, if we have the specific float and integer tests?

**Review by Bodacious — APPROVED** [review]
Nice one! Really happy with how the operator tests have turned out.

File: test/rubocop/cop/magic_numbers/no_argument/method_test.rb
Comment (AgentAntelope) [reply test/rubocop/cop/magic_numbers/no_argument/method_test.rb:1]: I like it because it confirms that it works for the `All` case

**Pull request merged**